src/monattack.cpp

```diff
@@ -185,6 +185,94 @@ void mattack::shriek(monster *z, int index)
     sounds::sound(z->pos(), 50, _("a terrible shriek!"));
 }
 
+void mattack::shriek_alert(monster *z, int index)
+{
+    if( !z->can_act() || z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+
+    if(g->u.sees( *z )){
+    add_msg( _("The %s begins shrieking!"), z->name().c_str());
+    }
+
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 15 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->moves -= 150;
+    z->reset_special(index); // Reset timer
+    sounds::sound(z->pos(), 120, _("a piercing wail!"));
+    z->add_effect("shrieking", 10);
+
+}
+
+static int posp( int posc ) //Determines orientation of target from the source
+{
+    if( posc > 0 ) {
+        return 1;
+    }
+    if( posc < 0 ) {
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+void mattack::shriek_stun(monster *z, int index)
+{
+    if( !z->can_act() || !z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 7 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->reset_special(index); // Reset timer
+
+    float dx = posp( target->posx() - z->posx() );
+    float dy = posp( target->posy() - z->posy() ); //Find direction of player
+    int i = 0; int j = 0;
+    int bsize = 0;  //Set range of our boxes
+    for( int range = 0; range < 4; range++ ) {
+        dx = dx * 1.5;
+        dy = dy * 1.5;
+        i++;  j++; bsize--; //Increase size of boxes after each distance toward target
+        for (int boxx = bsize; boxx <= i; boxx++) {
+            for (int boxy = bsize; boxy <= j; boxy++) {
+                tripoint cone( z->posx() + dx + boxx,
+                               z->posy() + dy + boxy,
+                               z->posz() );
+            if( z->sees(cone) ) {
+                g->m.bash( cone, 3, true ); //Small bash to every square, silent to not flood message box
+                if (g->mon_at(cone) != -1){
+                    const int mondex = g->mon_at(cone);
+                    Creature *zed = &g->zombie( mondex ); //If a monster is there, chance for stun
+                    if ( one_in((dist + 8 )) && !(zed->is_immune_effect("deaf")) && !zed->has_effect("stunned") ){
+                        zed->add_effect("stunned", rng(2,3));
+
+                    }
+                } else if (cone == g->u.pos() || cone == target->pos()){ //Same for player
+                    if ( one_in((dist + 8 )) && !(target->is_immune_effect("deaf")) && !target->has_effect("stunned") ){
+                        target->add_effect("stunned", rng(2,3));
```

Comment:
Dispersion on ranged weapons doesn't really matter at that range.
There are 3 ranges to consider: point blank (dispersion doesn't matter unless it gets huge), sniping (won't be done when zeds are around) and medium (which is only really used vs very tough targets like early predators, hulks or shocker brutes).

Unless you want to give it a really good range, it probably won't have much of a chance to affect a ranged attacker.

Lowering perception and dexterity would indirectly lower ranged accuracy, though it could hurt the low-dex survivors more than the high-dex ones. Lowering only the perception would only affect ranged survivors trying to attack at medium range, which doesn't happen often.

Increasing recoil would make sense as it would affect the point-blank shooting too, but it would require the attack to occur every single turn to be effective. Or be hardcoded.

Speed debuff could stack dangerously with grabbed speed penalty. Unfortunately speed debuffs, being linear, get more dangerous the lower your speed is.

Not sure what should be the most proper debuff here.